test/preview/document_stores/test_memory.py

```diff
@@ -12,3 +15,123 @@ class TestMemoryDocumentStore(DocumentStoreBaseTests):
+    @pytest.mark.unit
+    def test_bm25_retrieval_with_different_top_k(self, docstore):
+        # Tests if the bm25_retrieval method correctly changes the number of returned documents
+        # based on the top_k parameter.
+        docs = [
+            Document.from_dict({"content": "Hello world"}),
+            Document.from_dict({"content": "Haystack supports multiple languages"}),
+            Document.from_dict({"content": "Python is a popular programming language"}),
+        ]
+        docstore.write_documents(docs)
+
+        # top_k = 2
+        results = docstore.bm25_retrieval(query="languages", top_k=2)
+        assert len(results) == 2
+
+        # top_k = 3
+        results = docstore.bm25_retrieval(query="languages", top_k=3)
+        assert len(results) == 3
+
+    # Test two queries and make sure the results are different
+    @pytest.mark.unit
+    def test_bm25_retrieval_with_two_queries(self, docstore):
+        # Tests if the bm25_retrieval method returns different documents for different queries.
+        docs = [
+            Document.from_dict({"content": "Javascript is a popular programming language"}),
+            Document.from_dict({"content": "Java is a popular programming language"}),
+            Document.from_dict({"content": "Python is a popular programming language"}),
+            Document.from_dict({"content": "Ruby is a popular programming language"}),
+            Document.from_dict({"content": "PHP is a popular programming language"}),
+        ]
+        docstore.write_documents(docs)
+
+        results1 = docstore.bm25_retrieval(query="Java", top_k=1)
+        results2 = docstore.bm25_retrieval(query="Python", top_k=1)
+        assert results1[0].content == "Java is a popular programming language"
+        assert results2[0].content == "Python is a popular programming language"
+
+    # Test a query, add a new document and make sure results are appropriately updated
+    @pytest.mark.unit
+    def test_bm25_retrieval_with_updated_docs(self, docstore):
+        # Tests if the bm25_retrieval method correctly updates the retrieved documents when new
+        # documents are added to the store.
+        docs = [Document.from_dict({"content": "Hello world"})]
+        docstore.write_documents(docs)
+
+        results1 = docstore.bm25_retrieval(query="Python", top_k=1)
+        assert len(results1) == 1
+
+        docstore.write_documents([Document.from_dict({"content": "Python is a popular programming language"})])
+        results2 = docstore.bm25_retrieval(query="Python", top_k=1)
+        assert len(results2) == 1
+        assert results2[0].content == "Python is a popular programming language"
+
+        docstore.write_documents([Document.from_dict({"content": "Java is a popular programming language"})])
+        results3 = docstore.bm25_retrieval(query="Python", top_k=1)
+        assert len(results3) == 1
+        assert results3[0].content == "Python is a popular programming language"
```

Comment:
We should reuse `results`, makes it easier to read.

```suggestion
        results = docstore.bm25_retrieval(query="Python", top_k=1)
        assert len(results) == 1

        docstore.write_documents([Document.from_dict({"content": "Python is a popular programming language"})])
        results = docstore.bm25_retrieval(query="Python", top_k=1)
        assert len(results) == 1
        assert results[0].content == "Python is a popular programming language"

        docstore.write_documents([Document.from_dict({"content": "Java is a popular programming language"})])
        results = docstore.bm25_retrieval(query="Python", top_k=1)
        assert len(results) == 1
        assert results[0].content == "Python is a popular programming language"
```